libs/agno/agno/document/reader/reader_factory.py

```diff
@@ -0,0 +1,238 @@
+            "name": "PDF Reader",
+            "description": "Reads PDF files",
+        },
+        "csv": {
+            "class": "agno.document.reader.csv_reader.CSVReader",
+            "config": {"name": "CSV Reader", "description": "Reads CSV files"},
+            "extensions": [".csv"],
+            "name": "CSV Reader",
+            "description": "Reads CSV files",
+        },
+        "docx": {
+            "class": "agno.document.reader.docx_reader.DocxReader",
+            "config": {"name": "Docx Reader", "description": "Reads Docx files"},
+            "extensions": [".docx", ".doc"],
+            "name": "Docx Reader",
+            "description": "Reads Docx files",
+        },
+        "json": {
+            "class": "agno.document.reader.json_reader.JSONReader",
+            "config": {"name": "JSON Reader", "description": "Reads JSON files"},
+            "extensions": [".json"],
+            "name": "JSON Reader",
+            "description": "Reads JSON files",
+        },
+        "markdown": {
+            "class": "agno.document.reader.markdown_reader.MarkdownReader",
+            "config": {"name": "Markdown Reader", "description": "Reads Markdown files"},
+            "extensions": [".md", ".markdown"],
+            "name": "Markdown Reader",
+            "description": "Reads Markdown files",
+        },
+        "text": {
+            "class": "agno.document.reader.text_reader.TextReader",
+            "config": {"name": "Text Reader", "description": "Reads Text files"},
+            "extensions": [".txt", ".text"],
+            "name": "Text Reader",
+            "description": "Reads Text files",
+        },
+        # URL readers
+        "url": {
+            "class": "agno.document.reader.url_reader.URLReader",
+            "config": {"name": "URL Reader", "description": "Reads URLs"},
+            "name": "URL Reader",
+            "description": "Reads URLs",
+        },
+        "website": {
+            "class": "agno.document.reader.website_reader.WebsiteReader",
+            "config": {"name": "Website Reader", "description": "Reads Website files"},
+            "name": "Website Reader",
+            "description": "Reads Website files",
+        },
+        "firecrawl": {
+            "class": "agno.document.reader.firecrawl_reader.FirecrawlReader",
+            "config": {
+                "api_key": None,  # Will be set from env
+                "mode": "crawl",
+                "name": "Firecrawl Reader",
+                "description": "Crawls websites",
+            },
+            "name": "Firecrawl Reader",
+            "description": "Crawls websites",
+        },
+        "youtube": {
+            "class": "agno.document.reader.youtube_reader.YouTubeReader",
+            "config": {"name": "YouTube Reader", "description": "Reads YouTube videos"},
+            "name": "YouTube Reader",
+            "description": "Reads YouTube videos",
+        },
+        # URL file readers
+        "pdf_url": {
```

Comment:
Should this actually be a separate reader? We could just read the URL, then use the normal reader? Just thinking about it now to simplify